financialaid/tasks_test.py

```diff
@@ -3,14 +3,39 @@
 """
 from unittest.mock import patch
 
-from django.test import TestCase
+from django.test import TestCase, override_settings
 
-from financialaid.constants import CURRENCY_EXCHANGE_RATE_API_REQUEST_URL
 from financialaid.exceptions import ExceededAPICallsException, UnexpectedAPIErrorException
 from financialaid.models import CurrencyExchangeRate
 from financialaid.tasks import sync_currency_exchange_rates
 
 
+class TaskConfigurationTest(TestCase):
+    """
+    Test cases for configuration of financial aid tasks.
+    """
+    @override_settings(
+        OPEN_EXCHANGE_RATES_URL=None,
+        OPEN_EXCHANGE_RATES_APP_ID=None,
+    )
+    @patch('financialaid.tasks.CURRENCY_EXCHANGE_RATE_API_REQUEST_URL', None)
+    def test_unset_currency_exchange_api_url(self):
+        """
+        Assert that the task raises an exception if it is misconfigured.
+        """
+        with self.assertRaises(RuntimeError) as context:
+            sync_currency_exchange_rates()
+        assert "Currency exchange API URL cannot be determined" in str(context.exception)
+
+
+@override_settings(
+    OPEN_EXCHANGE_RATES_URL="https://openexchangerates.org/api/",
+    OPEN_EXCHANGE_RATES_APP_ID="fakeID123",
+)
+@patch(
+    'financialaid.tasks.CURRENCY_EXCHANGE_RATE_API_REQUEST_URL',
+    "https://openexchangerates.org/api/latest.json?app_id=fakeID123"
+)
```

Comment:
Because `CURRENCY_EXCHANGE_RATE_API_REQUEST_URL` isn't determined dynamically, it's determined once, when the Python application starts up. Changing the settings values won't automatically cause the constant to change.

Sowa rhr Mkw awnaw?